flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/java/BatchTableEnvironment.scala

```diff
@@ -178,4 +178,32 @@ class BatchTableEnvironment(
 
     registerTableFunctionInternal[T](name, tf)
   }
+
+  /**
+    * Scans a table from registered temporary tables and registered catalogs.
+    *
+    * The table to scan must be registered in the TableEnvironment or
+    * must exist in registered catalog in the TableEnvironment.
+    *
+    * Example:
+    *
+    * to scan a registered temporary table
+    * {{{
+    *   val tab: Table = tableEnv.scan("tableName")
+    * }}}
+    *
+    * to scan a table from a registered catalog
+    * {{{
+    *   val tab: Table = tableEnv.scan("catalogName.dbName.tableName")
+    * }}}
+    *
+    * @param tablePath The path of the table to scan.
+    * @throws TableException if no table is found using the given table path.
+    * @return The resulting [[Table]].
+    */
+  @throws[TableException]
+  def scan(tablePath: String): Table = {
```

Comment:
Why do we have two types of scan methods (one with string and one with var-args)?